File: frontend/src/types/supplier.ts
export type Category = 
  | 'Equipment' 
  | 'Consumables' 
  | 'Services' 
  | 'CRO_Contract' 
  | 'Salaries' 
  | 'Overheads' 
  | 'Travel' 
  | 'Other';

export type GstTreatment = 
  | 'CAP'  // Capital
  | 'EXP'  // Expense
  | 'FRE'  // GST Free
  | 'INP'  // Input Taxed
  | 'NTR'  // Not Reported
  | 'MIX'; // Mixed

export interface Supplier {
  id: string;
  name: string;
  abn: string | null;
  contact_name: string | null;
  contact_email: string | null;
  phone: string | null;
  address: string | null;
  gst_registered: boolean;
  default_gst_treatment: GstTreatment | null;
  default_category: Category | null;
  is_rd_supplier: boolean;
  notes: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface SupplierCreate {
  name: string;
  abn?: string | null;
  contact_name?: string | null;
  contact_email?: string | null;
  phone?: string | null;
  address?: string | null;
  gst_registered?: boolean;
  default_gst_treatment?: GstTreatment | null;
  default_category?: Category | null;
  is_rd_supplier?: boolean;
  notes?: string | null;
  is_active?: boolean;
}

export interface SupplierUpdate {
  name?: string;
  abn?: string | null;
  contact_name?: string | null;
  contact_email?: string | null;
  phone?: string | null;
  address?: string | null;
  gst_registered?: boolean;
  default_gst_treatment?: GstTreatment | null;
  default_category?: Category | null;
  is_rd_supplier?: boolean;
  notes?: string | null;
  is_active?: boolean;
}

export interface PaginatedSuppliers {
  items: Supplier[];
  total: number;
  page: number;
  page_size: number;
}

export const CATEGORY_OPTIONS: { value: Category; label: string }[] = [
  { value: 'Equipment', label: 'Equipment' },
  { value: 'Consumables', label: 'Consumables' },
  { value: 'Services', label: 'Services' },
  { value: 'CRO_Contract', label: 'CRO Contract' },
  { value: 'Salaries', label: 'Salaries' },
  { value: 'Overheads', label: 'Overheads' },
  { value: 'Travel', label: 'Travel' },
  { value: 'Other', label: 'Other' },
];

export const GST_TREATMENT_OPTIONS: { value: GstTreatment; label: string }[] = [
  { value: 'CAP', label: 'CAP - Capital' },
  { value: 'EXP', label: 'EXP - Expense' },
  { value: 'FRE', label: 'FRE - GST Free' },
  { value: 'INP', label: 'INP - Input Taxed' },
  { value: 'NTR', label: 'NTR - Not Reported' },
  { value: 'MIX', label: 'MIX - Mixed' },
];

File: backend/app/models/supplier.py
import uuid
from datetime import datetime
from sqlalchemy import Column, String, Boolean, DateTime, JSON, Enum, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
import enum

from app.database import Base


class Category(str, enum.Enum):
    EQUIPMENT = "Equipment"
    CONSUMABLES = "Consumables"
    SERVICES = "Services"
    CRO_CONTRACT = "CRO_Contract"
    SALARIES = "Salaries"
    OVERHEADS = "Overheads"
    TRAVEL = "Travel"
    OTHER = "Other"


class GstTreatment(str, enum.Enum):
    CAP = "CAP"  # Capital
    EXP = "EXP"  # Expense
    FRE = "FRE"  # GST Free
    INP = "INP"  # Input Taxed
    NTR = "NTR"  # Not Reported
    MIX = "MIX"  # Mixed


class Supplier(Base):
    __tablename__ = "suppliers"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    name = Column(String(255), nullable=False, index=True)
    abn = Column(String(20), nullable=True)
    contact_info = Column(JSON, default=dict)
    default_category = Column(Enum(Category), nullable=True)
    default_gst_treatment = Column(Enum(GstTreatment), nullable=True)
    notes = Column(String(1000), nullable=True)
    is_active = Column(Boolean, default=True, nullable=False)
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow, nullable=False)

    def __repr__(self):
        return f"<Supplier(id={self.id}, name={self.name})>"
